services/121-service/src/notifications/whatsapp/whatsapp.service.ts

```diff
@@ -73,51 +74,74 @@ export class WhatsappService {
       payload['messageContentType'] = messageContentType;
     }
 
-    let errorOccurred = false;
-    let messageToStore;
     try {
-      messageToStore = await twilioClient.messages.create(payload);
+      const messageToStore = await twilioClient.messages.create(payload);
+      await this.storeSendWhatsapp({
```

Comment:
The fact that this call was not there before my changes seems like a bug. The code in `finally` seems to assume that the `finally` clause is always called - but I believe that isn't true when there is a `return` statement in the `try` block.

Reply:
I actually tested this out for myself. So, the code from before seems to work - but by chance. With our mix of TS strictness and code, it could've easily broken if we ever added a `return` to the `finally` block (which, normally, TS would complain about).

Here is a [typescript playground](https://www.typescriptlang.org/play/?#code/IYZwngdgxgBAZgV2gFwJYHsL1RYAbPMAdVWQAsAlAU2QQCcIAKAShgG8AoGbmZOsdlx7ComEOjxUAdHnQBzRgHI4OfIRLlqtBr34BCRcwDcQ4dzo16WAIwnhAXxhRgyKGRiMqrTmZ6iI4pIy8koquATEpJSWOs6uZAbGpmb+gdKyCl52PI5hagI+vk5iEukhyqoRGtHaWHkRidm+FrUwAAxNMPYc3RwA9H0wAHJUVAAmvOgwwGMTwLwgALSochDoFjBkVBsARlTOCCBUADQwAO5RMCB8qFDIMAC26GNUMJinZ6-OEIr3a-ctKwwKgAN22YDOWws-UGAAFkEsVmtoaBILBECgMHVKuoougEMgtFYWIJhHwCskRCUghlQjjIuR8YSYlhyY1KTxATpbMlHHE3B4vKSiqlSsEFBVwrjGQSibEXG52UVigExbSsrzsFKKcrRTTyvVpWQmXLsdqlUUBjAuVgOryehwOIaGcbZSyWFJyFQmEyAA4EmAAXgAfMLuHqyhLndUTSzFKc-QSkslo1FTR6vT6Cf77iGwylqZG6drqqb429s0nOvYkjWjEA) which illustrates this.

I would argue that my code is therefore still an improvement, in that it is much more explicit.